url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/domain/singleagent/sokoban/SokobanVisualizer.java

```diff
@@ -0,0 +1,177 @@
+package domain.singleagent.sokoban;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.geom.Rectangle2D;
+import java.lang.reflect.Field;
+
+import burlap.oomdp.core.ObjectInstance;
+import burlap.oomdp.core.State;
+import burlap.oomdp.visualizer.ObjectPainter;
+import burlap.oomdp.visualizer.Visualizer;
+
+/**
+ * Javadoc test
+ * @author Richard Adjogah
+ *
+ */
+public class SokobanVisualizer {
+
+	/**
+	 * 
+	 * @return the visualizer object 
+	 */
+	public static Visualizer getVisualizer(){
+
+		Visualizer v = new Visualizer();
+
+
+		v.addObjectClassPainter(SokobanDomain.ROOMCLASS, new RoomPainter());
+		v.addObjectClassPainter(SokobanDomain.AGENTCLASS, new AgentPainter());
+		v.addObjectClassPainter(SokobanDomain.BLOCKCLASS, new BlockPainter());
+
+		return v;
+	}
+
+	//each invdividual object type should have its own painter classthat determines what color to draw and where
+	static class RoomPainter implements ObjectPainter{
+
+
+		@Override
+		public void paintObject(Graphics2D g2, State s, ObjectInstance ob,
+				float cWidth, float cHeight) {
+			
+			float domainXScale = (SokobanDomain.MAXX + 1) - SokobanDomain.MINX;
+			float domainYScale = (SokobanDomain.MAXY + 1) - SokobanDomain.MINY;
+			
+			//determines which java Color to use from a String. Defaults to Dark Gray if the string isn't a valid color
+			String color = ob.getStringValForAttribute(SokobanDomain.COLORATTNAME);
+			
+			Field field;
+			try {
+				field = Class.forName("java.awt.Color").getField(color);
+				g2.setColor((Color)field.get(null));
+
+			} catch (Exception e) {
+				g2.setColor(Color.darkGray);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 51:57, 155:161.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.